oc_mirror.sh

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env bash
+set -ex
+
+# Utility functions to create a local mirror registry using 'mirror-registry' and mirror
+# a release using 'oc mirror'
+
+function add_auth_to_pull_secret() {
+ 
+   quay_auths=`echo -n "${QUAY_REGISTRY_USER}:${QUAY_REGISTRY_PASS}" | base64 -w0`
+
+   tmpauthfile=$(mktemp --tmpdir "quayauth--XXXXXXXXXX")
+   _tmpfiles="$_tmpfiles $tmpauthfile"
+   tmppullsecret=$(mktemp --tmpdir "pullsecret--XXXXXXXXXX")
+   _tmpfiles="$_tmpfiles $tmppullsecret"
+
+
+   cat > "${tmpauthfile}" << EOF
+{
+  "auths": {
+    "${LOCAL_REGISTRY_DNS_NAME}:${LOCAL_REGISTRY_PORT}": {
+      "auth": "$quay_auths"
+    }
+  }
+}
+EOF
+
+   jq -s '.[0] * .[1]' ${tmpauthfile} ${PULL_SECRET_FILE} > ${tmppullsecret}
+   cp ${tmppullsecret} ${PULL_SECRET_FILE}
+
+   if [[ -f ${DOCKER_CONFIG_FILE} ]]; then
+      tmpdockerconfig=$(mktemp --tmpdir "dockerconfig--XXXXXXXXXX")
+      _tmpfiles="$_tmpfiles $tmpdockerconfig"
+
+      cp ${DOCKER_CONFIG_FILE} ${DOCKER_CONFIG_FILE}.old
+      jq -s '.[0] * .[1] * .[2]' ${PERSONAL_PULL_SECRET} ${tmpauthfile} ${DOCKER_CONFIG_FILE} > ${tmpdockerconfig}
+      cp ${tmpdockerconfig} ${DOCKER_CONFIG_FILE}
+      echo "In order to use the mirror registry, entries have  been added to ${DOCKER_CONFIG_FILE}"
+   else
+      echo "The ${DOCKER_CONFIG_FILE} does not exist, access to the local registry may fail authorization"
+   fi
+
+}
+
+function setup_quay_mirror_registry() {
+
+   quayHostName="virthost.ostest.test.metalkube.org"
+   dockerConfigFile=$HOME/.docker/config.json
+
+   mkdir -p ${WORKING_DIR}/oc-mirror
+   pushd ${WORKING_DIR}/mirror-registry
+   sudo ./mirror-registry install --quayHostname ${quayHostName} --quayRoot ${WORKING_DIR}/oc-mirror/ --initUser ${QUAY_REGISTRY_USER} --initPassword ${QUAY_REGISTRY_PASS} --sslCheckSkip -v
+
+   add_auth_to_pull_secret
+   popd
+
+}
+
+function publish_mirror_registry() {
+
+   # Create imageset containing the local URL and the OCP release to mirror
+   tmpimageset=$(mktemp --tmpdir "imageset--XXXXXXXXXX")
+   _tmpfiles="$_tmpfiles $tmpimageset"
+
+   cat > "${tmpimageset}" << EOF
+apiVersion: mirror.openshift.io/v1alpha2
+kind: ImageSetConfiguration
+archiveSize: 4
+storageConfig:
+  registry:
+    imageURL: ${LOCAL_REGISTRY_DNS_NAME}:${LOCAL_REGISTRY_PORT}/origin:latest
+    skipTLS: true
+mirror:
+  platform:
+    channels:
+    - name: candidate-${OPENSHIFT_RELEASE_STREAM}
+      type: ocp
+  additionalImages:
+  - name: registry.redhat.io/ubi8/ubi:latest
+EOF
+
+   oc mirror --dest-skip-tls --config ${tmpimageset} docker://${LOCAL_REGISTRY_DNS_NAME}:${LOCAL_REGISTRY_PORT}
```

Comment:
IIUC all the executions of `oc mirror` are performed in the current directory, while the working dir should be set to `${WORKING_DIR}` (preferably a sub folder) (I see the `oc-mirror.log` and `oc-mirror-workspace` folder being created in the curr dir, which was something not expected)